pytorch_lightning/utilities/warnings.py

```diff
@@ -14,17 +14,25 @@
 """Warning-related utilities."""
 import warnings
 from functools import partial
+from typing import Any, Union
 
 from pytorch_lightning.utilities.distributed import rank_zero_only
 
 
-def _warn(*args, stacklevel: int = 2, **kwargs):
-    warnings.warn(*args, stacklevel=stacklevel, **kwargs)
```

Comment:
Mypy doesn't want us passing `*args` like this because it doesn't conform to their typing stub (???) so I had to remove it.

This means we lose support for passing the category positionally, so I added a deprecation path for it